File: Lab5/RecyclerViewLab/app/src/main/java/mobiledev/unb/ca/recyclerviewlab/model/Course.java
package mobiledev.unb.ca.recyclerviewlab.model;

public class Course {

    private final String id;
    private final String name;
    private final String description;

    private Course (CourseBuilder builder) {
        this.id = builder.id;
        this.name = builder.name;
        this.description = builder.description;
    }

    /*public Course(String id, String name, String desc) {
        mId = id;
        mName = name;
        mDesc = desc;
    }*/

    public String getTitle() {
        return id + ": " + name;
    }

    public String getDescription() {
        return description;
    }

    public static class CourseBuilder {
        private String id;
        private String name;
        private String description;

        public CourseBuilder() {

        }

        public CourseBuilder id(String id) {
            this.id = id;
            return this;
        }

        public CourseBuilder name(String name) {
            this.name = name;
            return this;
        }

        public CourseBuilder description(String description) {
            this.description = description;
            return this;
        }

        public Course build() {
            return new Course(this);
        }
    }
}

File: Winter2022/labs/kotlin/Lab5/RecyclerViewLab/app/src/main/java/mobiledev/unb/ca/recyclerviewlab/model/Course.kt
package mobiledev.unb.ca.recyclerviewlab.model

class Course private constructor() {

    class Builder(
        private var id: String,
        private var name: String,
        private var description: String? = null
    ) {

        // Only need to include getters
        fun getTitle(): String {
            return "$id: $name"
        }

        fun getDescription(): String? {
            return description
        }

        fun build(): Course {
            return Course()
        }
    }
}
